consensus/src/persistent_liveness_storage.rs

```diff
@@ -395,6 +396,8 @@ impl<T: Payload> PersistentLivenessStorage<T> for StorageWriteProxy {
                 LivenessStorageData::RecoveryData(initial_data)
             }
             Err(e) => {
+                // @REVIEW: It seems like on this path you can potentially be byzantine (e.g., forget
+                // a previous vote). Should we allow this to happen without operator intervention?
```

Comment:
you can't be byzantine because of tcb. this is for the convenience when someone deletes consensusdb. I don't have strong opinion on this, this does add more complexity to the code.